### PR TITLE
Change: Use mocked SnapshotProvider

### LINKED ISSUE #1431: Use mocks for SnapshotProvider in unit tests
*Note*
The feature request will probably be integrated faster if you do a pull request for it.
If you want to discuss the feature before you actually write the code you are welcome to do it by first submitting an issue.

### Description
`SnapshotProviderImpl is used without a mock in various tests. We would like to mock it.
### Motivation
Using the actual provider slows down the unit tests. It also hurts the isolation principle.
### Requirements
Go over all the tests that use the `SanpshotProvider` and mock it

### PR DESCRIPTION
**Pull request by lukazd**
# Description

Replaces all real usages of SnapshotProvider/SnapshotProviderImpl in tests with mocks.

1. Mocks all usages of the SnapshotProvider in tests (except SnapshotProviderImplTest.java and the benchmark tests). 
2. Adds `createSnapshot()` methods to SnapshotMockUtils.java to create real snapshots to be used directly in tests or for stubbing a SnapshotProvider.
3. Removes SnapshotProvider entirely in tests where the class under test does not use the provider but just a Snapshot. A real snapshot object is used directly instead.

Fixes #1431

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

All tests continue to pass.

# Checklist:

_Please delete items that are not relevant._

- [X] My code follows the style guidelines for this project
- [X] I have performed a self-review of my own code
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/test/java/com/iota/iri/TransactionValidatorTest.java [hunk 1]
```diff
@@ -1,58 +1,68 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
 
 import static com.iota.iri.TransactionTestUtils.*;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/TransactionValidatorTest.java]: Can you replace this by adding:
```
  @Rule
    public MockitoRule mockitoRule = MockitoJUnit.rule();
```

The reason is this is the way we currently do it in the project.
I personally think that `@RunWith(MockitoJUnitRunner.class)` is nicer  :-)
But, we don't want to change it in all the unit tests in the project

Comment (lukazd) [reply src/test/java/com/iota/iri/TransactionValidatorTest.java]: no problem. I’ll switch it.

Comment (GalRogozinski) [reply src/test/java/com/iota/iri/TransactionValidatorTest.java]: are you doing it?

Comment (lukazd) [reply src/test/java/com/iota/iri/TransactionValidatorTest.java]: Yep, finally found some time to get to it. Changed in all places I added the runner before.

Comment (GalRogozinski) [reply src/test/java/com/iota/iri/TransactionValidatorTest.java]: I thought you were going to resolve conflicts as well.
Do you intend to?

Comment (lukazd) [reply src/test/java/com/iota/iri/TransactionValidatorTest.java]: You are right, just did it. Fixed conflicts and tests still passing.

**Pull request merged**